pages/querying/read-and-modify-data.md

```diff
@@ -532,3 +532,149 @@ relationships:
+stay the same when updates are done, or go down if deletes are performed.
+
+Another example of an unusual spike in memory is during the creation of graph entities in the graph.
+Consider the following query:
+
+```cypher
+LOAD CSV FROM "/temp.csv" WITH HEADER AS row
+CREATE (:Node {id: row.id, name: row.name});
+```
+
+This command will create multiple Delta for every CSV row ingested:
+- 1 delta object for creating a node in the graph
+- 1 delta object for adding a label `Node`
+- 2 delta objects for adding properties `id` and `name`
+
+The memory overhead could be even larger than the delete example below, since this query is generating four
+Delta objects per row.
+
+For these purposes, periodic execution can be used to batch the query into smaller chunks, which enables
+the Delta objects to be also deallocated during query execution. The final benefit for the user is having
+a lot smaller amount of Delta objects allocated at a point in time (in the magnitude of thousands, and not
+in millions).
+
+### USING PERIODIC COMMIT <num_rows>
+
+`USING PERIODIC COMMIT num_rows` is a pre-query directive which one can use to specify how often
+will the query be batched. After each batch, the system will attempt to clear the Delta objects
+to release additional memory to the system. This will succeed if this query is the only one executing in
+the system at the time. The `num_rows` literal is a non-negative integer which signifies after how 
+many rows will the Delta objects be cleaned from the system. 
+
+Consider again the following `LOAD CSV` example, but with a modification
+
+```cypher
+USING PERIODIC COMMIT 1000
+LOAD CSV FROM "/temp.csv" WITH HEADER AS row
+CREATE (:Node {id: row.id, name: row.name});
+```
+
+The query execution plan for this query looks like this:
+
+```
++---------------------+
+| QUERY PLAN          |
++---------------------+
+| " * EmptyResult"    |
+| " * PeriodicCommit" |
+| " * CreateNode"     |
+| " * LoadCsv {row}"  |
+| " * Once"           |
++---------------------+
+```
+
+The query plan is read from bottom to top. Notice the `PeriodicCommit` operator at the end of the query used to count the number of rows produced through
+that operator. After the counter reaches the limit (in this case 1000), the query will be batched and committed, 
+and the Delta objects will potentially be released, if the query is the only one executing at the time.
+
+Since in the above paragraph we have said that we create four Delta objects per row in the query, and with a Delta
+objects consisting of about 56 bytes, we can expect around 224KB of additional memory overhead during query execution,
+rather than 2.24GB if there are 10 million entries in the CSV.
+
+
+### CALL { subquery } IN TRANSACTIONS OF num_rows ROWS
+
+The `CALL { subquery } IN TRANSACTIONS OF num_rows ROWS` is an additional syntax for periodic execution. The
+`num_rows` literal is the non-negative integer which specifies after how many pulls of the input branch will the

```

Comment:
s/non-negative/positive

Reply:
s/after how many pulls of the input branch will the query be batched/the batch size.